### ISSUE TITLE
MSConvert: Demultiplexing gives mzML and mzXML output that violate the schema/spec

### ISSUE THREAD

**Issue by austinkeller** [ProteoWizard/pwiz#359]
### Context
Brian Searle reported that OpenSWATH interprets the wrong isolation scheme of demuxed files. This appears to be because the selected ion m/z is not updated. Brian confirmed that manually updating the selected ion m/z values with centers corresponding to the synthetic demuxed windows fixes the import issue for OpenSWATH.

Here is an example of a line of the mzML file that should be updated:
```
<cvParam cvRef="MS" accession="MS:1000744" name="selected ion m/z" value="500.4774" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
```

Brian provided an awk script written to repair his mzML:
```
#!/usr/bin/awk -f
{
	if ($0 ~ /<processingMethod order=\"2\">/) {
		# fix the missing softwareRef attribute in prism
		print "        <processingMethod order=\"2\" softwareRef=\"pwiz\">";

	} else if ($0 ~ /<spectrum index=/) {
		# fix the scan numbers so they are consecutive, remember to save them in ids[]
		current=$0;
		split($0, a, "\"");
		num=a[2]+1;
		id=a[4];
		split(id, b, "scan=");
		idpre=b[1];
		newid=idpre"scan="num;
		ids[id]=newid;
		print a[1] "\"" a[2] "\"" a[3] "\"" newid "\"" a[5] "\"" a[6] "\"" a[7];

	} else if ($0 ~ /accession=\"MS:1000827\"/) {
		# grab the correct scan center, save it as target
		split($0, a, "\"");
		target=a[8];
		print $0;

	} else if ($0 ~ /accession=\"MS:1000744\"/) {
		# insert the previous target to correct the incorrect center
		split($0, a, "\"");
		print a[1] "\"" a[2] "\"" a[3] "\"" a[4] "\"" a[5] "\"" a[6] "\"" a[7] "\"" target "\"" a[9] "\"" a[10] "\"" a[11] "\"" a[12] "\"" a[13] "\"" a[14] "\"" a[15];

	} else if ($0 ~ /<precursor spectrumRef=/) {
		# fix the scan index to reflect the new scan numbers
		# for some reason, these DO NOT include the "demux=0" tag. Thanks Austin!
		split($0, a, "\"");
		key=a[2]" demux=0";
		print a[1] "\"" ids[key] "\"" a[3];

	} else if ($0 ~ /<offset idRef=/) {
		# fix the scan index to reflect the new scan numbers
		split($0, a, "\"");
		print a[1] "\"" ids[a[2]] "\"" a[3];

	} else {
		print $0;
	}
}
```

Import of mzXML's also failed in other tools that do schema validation during import.

### Problem
The demultiplexing filter fails to uphold the mzML and mzXML schema specs. Because pwiz uses mzML as it's internal model of data, fixing the mzML schema issues should be the first step. I expect this to fix the mzXML issues as well.

### Potential Solution
Run an output file from demultiplexing through an mzML schema validator and fix any issues. Then verify that no issues remain by converting to mzXML and running through an mzXML schema validator.

**Comment by austinkeller** [comment]
Beginning work in https://github.com/austinkeller/pwiz/tree/feature/fix-demux-mzXML-schema

**Comment by austinkeller** [comment]
Splitting some pre-existing schema validation issues with mzXML into their own threads:
- https://github.com/ProteoWizard/pwiz/issues/384

**Comment by austinkeller** [comment]
Resolved by pull request #390
